opsdroid/connector/slack/__init__.py

```diff
@@ -70,85 +63,65 @@ def __init__(self, config, opsdroid=None):
             await self.disconnect()
             raise
 
-    async def reconnect(self, delay=None):
-        """Reconnect to the websocket."""
-        try:
-            self.reconnecting = True
-            if delay is not None:
-                await asyncio.sleep(delay)
-            await self.connect()
-        finally:
-            self.reconnecting = False
-
     async def disconnect(self):
         """Disconnect from Slack."""
-        await self.slacker.close()
+        self.listening = False
+        await self.slack_rtm.stop()
 
     async def listen(self):
         """Listen for and parse new messages."""
-        while self.listening:
-            try:
-                await self.receive_from_websocket()
-            except AttributeError:
-                break
-
-    async def receive_from_websocket(self):
-        """Get the next message from the websocket."""
-        try:
-            content = await self.websocket.recv()
-            await self.process_message(json.loads(content))
-        except websockets.exceptions.ConnectionClosed:
-            _LOGGER.info("Slack websocket closed, reconnecting...")
-            await self.reconnect(5)
 
-    async def process_message(self, message):
+    async def process_message(self, **payload):
         """Process a raw message and pass it to the parser."""
-        if "type" in message and message["type"] == "message" and "user" in message:
-
-            # Ignore bot messages
-            if "subtype" in message and message["subtype"] == "bot_message":
-                return
-
-            # Lookup username
-            _LOGGER.debug("Looking up sender username")
-            try:
-                user_info = await self.lookup_username(message["user"])
-            except ValueError:
-                return
-
-            # Replace usernames in the message
-            _LOGGER.debug("Replacing userids in message with usernames")
-            message["text"] = await self.replace_usernames(message["text"])
-
-            await self.opsdroid.parse(
-                Message(
-                    message["text"],
-                    user_info["name"],
-                    message["channel"],
-                    self,
-                    raw_event=message,
-                )
+        message = payload["data"]
+
+        # Ignore own messages
+        if "bot_id" in message and message["username"] == self.bot_name:
+            return
+
+        # Lookup username
+        _LOGGER.debug("Looking up sender username")
+        try:
+            user_info = await self.lookup_username(message["user"])
+        except ValueError:
+            return
+
+        # Replace usernames in the message
+        _LOGGER.debug("Replacing userids in message with usernames")
+        message["text"] = await self.replace_usernames(message["text"])
```

Comment:
oh this is going to conflict with #1071 :rocket:

Reply:
Yeah sorry there may be a few breakages with that PR.

Reply:
I am sure it wont be that hard to unravel.